src/docfx/build/toc/TableOfContentsMap.cs

```diff
@@ -11,16 +11,23 @@ namespace Microsoft.Docs.Build
     /// </summary>
     internal class TableOfContentsMap
     {
-        private readonly List<Document> _tocs;
+        private readonly HashSet<Document> _tocs;
 
         private readonly Dictionary<Document, HashSet<Document>> _documentToTocs;
 
         public TableOfContentsMap(List<Document> tocs, Dictionary<Document, HashSet<Document>> documentToTocs)
         {
-            _tocs = tocs ?? throw new ArgumentNullException(nameof(tocs));
+            _tocs = new HashSet<Document>(tocs ?? throw new ArgumentNullException(nameof(tocs)));
```

Comment:
debug.assert instead of throw?

Reply:
also assert `_toc.Count == tocs.Count`

Reply:
sorry, missing your comment, will change it later